### ISSUE TITLE
In DITA-OT project files, do not quietly ignore an output.dir <param> definition

### ISSUE THREAD

**Issue by chrispy-snps** [dita-ot/dita-ot#3913]
## Description
When the `output.dir` parameter is specified on a `dita` command line:

```
dita -i map.ditamap -f html5 -Doutput.dir=foo
Error: Property output.dir cannot be set with -D, use -o instead
```

then the DITA-OT warns that the parameter cannot be set directly and the `--output` option should be used instead:

But when a DITA-OT project file specifies the `output.dir` parameter:

```
<publication transtype="html5">
  <param name="output.dir" path="foo"/>
</publication>
```

then the parameter is quietly ignored, potentially resulting in user confusion.

## Possible Solution

Issue a similar warning/error message that the `output.dir` parameter cannot be defined in that context.

## Potential Alternatives
Consider the `output.dir` parameter somehow, although precedence with (or concatenation with??) the `--output` command-line option must be considered.

## Testcase
[ditaot_project_output_dir_ignored.zip](https://github.com/dita-ot/dita-ot/files/8472392/ditaot_project_output_dir_ignored.zip)

**Comment by jelovirt** [comment]
The fix itself is pretty simple, we can filter unsupported parameters in `Project.build()`. However, it may make sense to filterer parameters later on in the process, because we don't have a `Logger` in `Project` right now. Maybe if we moved `Project.build()` into `ProjectBuilder.build()`, which would make sense in general.

**Comment by chrispy-snps** [comment]
Fixed via #3915. Thanks @jelovirt!
